google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -197,4 +212,72 @@ public AppEngineStandardRuntime getAppEngineStandardDeclaredRuntime(
     }
   }
 
+  @Override
+  public Optional<String> getServiceNameFromAppYaml(@NotNull String appYamlPathString) {
+    return getValueFromAppYaml(appYamlPathString, SERVICE_TAG_YAML);
+  }
+
+  @Override
+  public FlexibleRuntime getFlexibleRuntimeFromAppYaml(@NotNull String appYamlPathString) {
+    return getValueFromAppYaml(appYamlPathString, RUNTIME_TAG_YAML).map(runtimeValue -> {
+      if (runtimeValue.equals(JAVA_RUNTIME)) {
+        return FlexibleRuntime.JAVA;
+      }
+      if (runtimeValue.equals(CUSTOM_RUNTIME)) {
+        return FlexibleRuntime.CUSTOM;
+      }
+      return FlexibleRuntime.UNKNOWN;
+    }).orElse(FlexibleRuntime.UNKNOWN);
+  }
+
+  private Optional<String> getValueFromAppYaml(@NotNull String appYamlPathString,
+      @NotNull String key) {
+    Path appYamlPath;
+    try {
+      appYamlPath = Paths.get(appYamlPathString);
+    } catch (InvalidPathException ipe) {
+      return Optional.empty();
+    }
+
+    try {
+      return Files.readAllLines(appYamlPath).stream()
+          .filter(line -> line.startsWith(key))
+          .map(line -> line.substring(key.length()).trim())
+          .findFirst();
+    } catch (IOException ioe) {
+      return Optional.empty();
+    }
+  }
+
+  @Override
+  public String getServiceNameFromAppEngineWebXml(
+      Project project, DeploymentSource deploymentSource) {
+    XmlFile appengineWebXml = loadAppEngineStandardWebXml(project, deploymentSource);
```

Comment:
I didn't know that class existed. Although you've filed an issue above to move this kind of parsing to AppEngineStandardWebApp?

Is any other issue required here? e.g., Create an AppEngineFlexibleWebApp class?